// @sidebase/nuxt-pdf/dist/module.mjs
import { defineNuxtModule, createResolver, addImportsDir } from '@nuxt/kit';
import { defu } from 'defu';

const PACKAGE_NAME = "@sidebase/nuxt-pdf";
const defaultOptions = {
  defaultDocOptions: {
    size: "A4",
    bufferPages: true,
    margins: {
      top: 25,
      left: 25,
      right: 25,
      bottom: 25
    }
  }
};
const module = defineNuxtModule({
  meta: {
    name: PACKAGE_NAME,
    configKey: "pdf"
  },
  // Default configuration options of the Nuxt module
  defaults: defaultOptions,
  setup(options, nuxt) {
    const resolver = createResolver(import.meta.url);
    nuxt.options.runtimeConfig.public.pdf = options;
    nuxt.hook("nitro:config", (nitroConfig) => {
      nitroConfig.alias = nitroConfig.alias || {};
      nitroConfig.externals = defu(
        typeof nitroConfig.externals === "object" ? nitroConfig.externals : {},
        {
          inline: [resolver.resolve("./runtime")]
        }
      );
      nitroConfig.alias["#pdf"] = resolver.resolve("./runtime/server");
    });
    const composables = resolver.resolve("./runtime/composables");
    addImportsDir(composables);
  }
});

export { module as default };
